services/roomService/src/client/CoveyTownsSocket.test.ts

```diff
@@ -77,6 +78,21 @@ describe('TownServiceApiSocket', () => {
     expect(movedPlayer.location).toMatchObject(newLocation);
     expect(otherMovedPlayer.location).toMatchObject(newLocation);
   });
+  it('Dispatches town messages to all clients in the same town, including the client who sent the message', async () => {
+    const town = await createTownForTesting();
+    const joinData = await apiClient.joinTown({coveyTownID: town.coveyTownID, userName: nanoid()});
+    const joinData2 = await apiClient.joinTown({coveyTownID: town.coveyTownID, userName: nanoid()});
+    const joinData3 = await apiClient.joinTown({coveyTownID: town.coveyTownID, userName: nanoid()});
+    const {socket, messageReceived} = TestUtils.createSocketClient(server, joinData.coveySessionToken, town.coveyTownID);
+    const {messageReceived: messageReceived2} = TestUtils.createSocketClient(server, joinData2.coveySessionToken, town.coveyTownID);
+    const {messageReceived: messageReceived3} = TestUtils.createSocketClient(server, joinData3.coveySessionToken, town.coveyTownID);
+    const message = TestUtils.createMessageForTesting(MessageType.TownMessage, new Player(nanoid()));
+    socket.emit('messageSent', message);
+    const [firstMessage, secondMessage, thirdMessage] = await Promise.all([messageReceived, messageReceived2, messageReceived3]);
+    expect(firstMessage).toMatchObject(message);
+    expect(secondMessage).toMatchObject(message);
+    expect(thirdMessage).toMatchObject(message);
+  });
```

Comment:
I think we should at least test whether the proximity and dm chats promises resolved for cases where they should resolve. worried about getting points off for not testing thoroughly enough. I'm also wondering if we could do a try catch block of some kind to still test:
```
try {
  await messageReceived2;
} catch (e) {
  expect(e).toBeTruthy
}

```
I think this _might_ work because the await ensures that it has a 5000ms timeout before giving up and then when the timeout error happens, it throws an error we're expecting

Reply:
@michelle-pine This sadly doesn't work -- the "error" thrown by Jest to notify of an async timeout isn't actually a catchable error, so it will still timeout. I'll still put up a test for resolvable promises for the proximity chat though!